content/Hardware Support/Alvik/Update-firmware-and-libraries-for-Arduino-Alvik.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Update firmware and libraries for Arduino Alvik"
+---
+
+Keeping your Alvik’s firmware updated ensures it has the latest features, bug fixes, and performance improvements. Regular updates also help maintain compatibility with other software, ensuring optimal performance for your robot. In this article, we will explain what the Alvik Updater does and how to use it effectively.
+
+- [About Alvik Updater](#about-alvik-updater)
+- [How to update your Alvik](#how-to-update-your-alvik)
+- [Troubleshooting](#troubleshooting)
+
+## About Alvik Updater
+
+**The Alvik Updater will:**
+
+- Update or install the Alvik MicroPython library
+- Update or install the Alvik Carrier firmware
+
+The Alvik Updater will not restore default demo examples, remove any projects or files added by the user, or update/install the MicroPython firmware on the Nano ESP32 board. For instructions on updating or installing the MicroPython firmware, refer to [the MicroPython firmware installation guide](https://support.arduino.cc/hc/en-us/articles/15156644171548-Upload-firmware-for-Arduino-Lab-for-Micropython).
+
+## How to update your Alvik
+
+1. Go to the [Arduino Alvik Updater](https://alvikupdate.arduino.cc/)
+
+1. Turn on your Alvik robot and connect the Nano ESP32 to your Computer with a USB-C cable.
+
+1. Click on the 'Connect' button.
+
+1. A dialog window will open, prompting you to select a device.
+
+   !["Connect to serial port" dialog window on PC and Chromebook](img/connect-to-serial-port.png)
+
+   > [!NOTE]
+   > The appearance of this window may vary depending on your operating system. Similarly, the port you need to select might differ in name or format.
+
+1. Once the device is connected, the 'Update' button will become clickable. Click the 'Update' button to proceed.
+
+1. Wait for the update to finish. This process may take some time.
+
+   ![Update Complete visuals](img/update-complete.png)
+
+   The update is complete when a checkmark and the text "Updated" appear below the update button.
+
+## Troubleshooting
+
+### If the update seems to have stopped
+
+If the update button seems unresponsive or does not continue loading:
+
+1. Ensure the switch on the Alvik is set to "On."
+
+1. Reload the page and try the update again.
+
+### If your Alvik doesn’t appear in the dialog window
+
+1. **Verify the board port in Arduino Lab for MicroPython**
+
+   Alvik is often not displayed as "Alvik" in the dialog window. It may sometimes appear as "Nano ESP32" or under another name, with the port listed in parentheses e.g., (COM29) on Windows or (ttyACM0) on Chromebook. If there are multiple options, you can confirm which port corresponds to your Alvik by connecting it to Arduino Lab for MicroPython.
```

Comment:
> If there are multiple options, you can confirm which port corresponds to your Alvik by connecting it to Arduino Lab for MicroPython.

I tested this on my sytem.

Advantages:
* Arduino Lab does not list virtual ports (Bluetooth, wlan)

Limiations:
* Arduino Labs will list other USB serial devices, such as other Arduino boards.
* Arduino Labs displays any additional information that lets me identify the port of a Alvik or Nano ESP32

So this recommendation may not work for all users.

With Chrome on macOS, the serial port dialog will update whenever there is a change port availability. So the quickest way for me to verify which port belongs to my Alvik is to disconnect and reconnect the device. If you think this could work, maybe that tip can be incorporated somewhere (perhaps [here](https://github.com/arduino/help-center-content/pull/462/files#diff-a6c8623b7cecad0d66c4422a0e449bcdf5563e4d81ba5aac06fa8fa89669b37eR33)?)